xlators/mount/fuse/src/fuse-bridge.c

```diff
@@ -6908,31 +6907,25 @@ init(xlator_t *this_xl)
         priv->congestion_threshold = priv->background_qlen;
     }
 
+    /* What you need to find with we do 'df -h' */
     cmd_args = &this_xl->ctx->cmd_args;
-    fsname = cmd_args->volfile;
-    if (!fsname && cmd_args->volfile_server) {
-        if (cmd_args->volfile_id) {
-            int dir_len = 0;
-            if (cmd_args->subdir_mount)
-                dir_len = strlen(cmd_args->subdir_mount) + 1;
-            fsname = GF_MALLOC(strlen(cmd_args->volfile_server) + 1 +
-                                   strlen(cmd_args->volfile_id) + 1 + dir_len,
-                               gf_fuse_mt_fuse_private_t);
-            if (!fsname) {
-                gf_log("glusterfs-fuse", GF_LOG_ERROR, "Out of memory");
-                goto cleanup_exit;
-            }
-            fsname_allocated = 1;
+    if (cmd_args->fs_display_name) {
+        strcpy(fsname, cmd_args->fs_display_name);
```

Comment:
`fs_display_name` comes from data passed by the user through command line, so it's unsafe. We shouldn't copy it to a stack allocated buffer without proper length checking.

Reply:
Ack